### PR TITLE
script for listing rpm versions from osg test runs

### PR DESCRIPTION
**Pull request by edquist**
This was taken largely from [`compare-rpm-versions`](https://github.com/opensciencegrid/tools/blob/master/compare-rpm-versions) script, which was really designed for comparing rpm versions between two different osg test runs (or two different root.logs, or two different rpm -qa outputs).

What it lacked (which this new script provides) is to simply list the versions of particular packages for a single run, or to summarize that information across all output dirs for a set of runs.

This is intended to be installed and run from osghost, but it can be run against local copies of data also (an entire osg test run-DATE dir, a single output-NNN dir, a root.log, or an rpm -qa output).

Time flies when you're up coding instead of sleeping...

### CODE REVIEW

File: list-rpm-versions [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        if strip_arch:
+            na = arch_strip(na)
+        if strip_dist:
+            evr = dist_strip(evr)
+        if evr.startswith("0:"):
+            evr = evr[2:]
+        yield [na,evr]
+        items[:2] = []
+
+def isdir(fn):
```
Comment (edquist) [inline list-rpm-versions]: Ah, looks like this started out in `compare-rpm-versions` as an [`isfile` function](https://github.com/edquist/tools/blob/7d49e71b633c30a03e195e37d6abfd1694c17feb/compare-rpm-versions#L71-L75), which probably there wasn't an equivalent of in `os.path` ... when I [changed the logic](
https://github.com/edquist/tools/commit/09275a443944997301ab16632a5027793db8b700#diff-1dbfd2157db6e355574163e895b4585eR75) to use `isdir` I made the obvious change without realizing there was an `os.path.isdir` that did the same thing.

**Comment by edquist** [comment]
Alright, this seems pretty usable on osghost.

I'm pasting some use cases for reference / appetite whetting.

TL;DR: The most common use case will probably be the one at the end with `--summarize` and `--list-outputs` (`-sl` for short) run against the timestamp for a VMU run dir.

---

Usage & Options summary:
```
[edquist@osghost ~]
$ list-rpm-versions --help

Usage:
  list-rpm-versions [options] output-001 [packages...]
  list-rpm-versions [options] --summarize [run-]20161220-1618 packages...

List version-release numbers for RPMs installed in an osg-test run output
directory, as found in output-NNN/output/osg-test-*.log

The output argument can also be a root.log from a koji/mock build,
or the raw output of an 'rpm -qa' command.

If any packages are specified, limit the results to just those packages.

If a run directory (or, just the timstamp string) is specified, summary
information will be printed for the listed packages across all output-NNN
subdirectories for that set of osg test runs.

Options:
  -A, --no-strip-arch  don't attempt to strip .arch from package names
  -D, --no-strip-dist  don't attempt to strip .dist tag from package releases

  -s, --summarize      summarize results for all output subdirs
  -l, --list-outputs   list output numbers (summarize mode only)
  -L, --max-outputs N  list at most N output numbers per NVR (-1 for unlimited)
```


Example run on a single `output-NNN` dir for all packages:
```
[edquist@osghost /osgtest/runs/run-20161221-0423]
$ list-rpm-versions output-123 

Package                         output-123
-------                         ----------
CGSI-gSOAP                      1.3.10-1
GConf2                          3.2.6-8
apache-commons-cli              1.2-13
apache-commons-codec            1.8-7
apache-commons-collections      3.2.1-22
apache-commons-discovery        2:0.5-9
apache-commons-io               1:2.4-12
apache-commons-lang             2.6-15
apache-commons-logging          1.1.2-7
apr                             1.4.8-3
apr-util                        1.5.2-6
atk                             2.14.0-1
audit-libs-python               2.4.1-5
avalon-framework                4.3-10
...
```


Example run on a single `output-NNN` dir for two packages:
```
[edquist@osghost /osgtest/runs/run-20161221-0423]
$ list-rpm-versions output-123 condor java-1.7.0-openjdk

Package             output-123
-------             ----------
condor              8.5.8-1.osgup
java-1.7.0-openjdk  1:1.7.0.121-2.6.8.0
```


Example run in summary mode over all `output-NNN` subdirs for a run set:
```
[edquist@osghost ~]
$ list-rpm-versions -s 20161221-0423 condor java-1.7.0-openjdk

Package             Version-Release      Count
-------             ---------------      -----
condor              -                    5
condor              8.4.9-1              63
condor              8.4.10-1             105
condor              8.5.7-1.osgup        42
condor              8.5.8-1.osgup        79

java-1.7.0-openjdk  -                    5
java-1.7.0-openjdk  1:1.7.0.121-2.6.8.0  121
java-1.7.0-openjdk  1:1.7.0.121-2.6.8.1  168
```


Same thing, but list the output dir numbers also:
```
[edquist@osghost ~]
$ list-rpm-versions -sl 20161221-0423 condor java-1.7.0-openjdk

Package             Version-Release      Count  Output-Nums
-------             ---------------      -----  -----------
condor              -                    5      075,078,080,082,083
condor              8.4.9-1              63     000,001,002,003,004,005,006,...
condor              8.4.10-1             105    007,008,009,010,011,012,013,...
condor              8.5.7-1.osgup        42     021,022,023,024,025,026,027,...
condor              8.5.8-1.osgup        79     028,029,030,031,032,033,034,...

java-1.7.0-openjdk  -                    5      075,078,080,082,083
java-1.7.0-openjdk  1:1.7.0.121-2.6.8.0  121    000,001,002,003,004,005,006,...
java-1.7.0-openjdk  1:1.7.0.121-2.6.8.1  168    126,127,128,129,130,131,132,...
```

**Comment by brianhlin** [comment]
The examples are really useful. Could we get them added to the `README` or somewhere else more easily visible?
